src/Microsoft.Health.Fhir.Core/Features/Resources/Get/GetResourceHandler.cs

```diff
@@ -3,33 +3,75 @@
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
+using System;
 using System.Threading;
 using System.Threading.Tasks;
 using EnsureThat;
+using Hl7.Fhir.Model;
 using MediatR;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Conformance;
 using Microsoft.Health.Fhir.Core.Features.Persistence;
 using Microsoft.Health.Fhir.Core.Messages.Get;
 
 namespace Microsoft.Health.Fhir.Core.Features.Resources.Get
 {
-    public class GetResourceHandler : IRequestHandler<GetResourceRequest, GetResourceResponse>
+    public class GetResourceHandler : BaseResourceHandler, IRequestHandler<GetResourceRequest, GetResourceResponse>
     {
-        private readonly IFhirRepository _repository;
+        private readonly IDataStore _dataStore;
```

Comment:
Is there any reason this and the arg check couldn't be in the `BaseResourceHandler`?